File: c++/menor posicao/menor posicao.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main(){
    int tamanho = 0;
    cin >> tamanho;

    int vetor[tamanho];

    int posicao = 0;
    int minimo = 0;

    for(int i = 0; i < tamanho; i++){
        cin >> vetor[i];
        
        if(minimo == 0){
            minimo = vetor[i];
        }

        if(vetor[i] < minimo){
            minimo = min(minimo, vetor[i]);
            posicao = i;
        }
    }

    cout << "Menor valor: " << minimo << endl;
    cout << "Posicao: " << posicao << endl;

    return 0;
}

File: java/menor posicao/MenorPosicao.java
import java.util.Scanner;

public class MenorPosicao {

    public void solution() {
        Scanner scanner = new Scanner(System.in);

        int size = scanner.nextInt();
        int position = 0;
        int minimum = 0;

        for (int i = 0; i < size; i++) {
            int input = scanner.nextInt();

            if (minimum == 0) {
                minimum = input;
            }

            if (input < minimum) {
                minimum = Math.min(minimum, input);
                position = i;
            }

        }

        scanner.close();

        System.out.println("Menor valor: " + minimum + "\n" + "Posicao: " + position);
    }

    public static void main(String[] args) {
        MenorPosicao menorPosicao = new MenorPosicao();
        menorPosicao.solution();
    }
}
